### PR TITLE
doc/devguide: Submission and style

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #4759

This commit adds code submission and coding style guidelines to the
devguide. Most of the material is a straight port from the wiki but
there have been some content modifications and additions.

Describe changes:
- Add TODO for empty sections

### CODE REVIEW

File: doc/devguide/code-style.rst [hunk 1]
```diff
@@ -1,2 +1,334 @@
...
+
+inline
+^^^^^^
+
+The inlining of functions should be used only in critical paths.
+
+curly braces / brackets
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Functions should have the opening bracket on a newline:
```
Comment (norg) [inline doc/devguide/code-style.rst:127]: isn't that already covered at the top?

File: doc/devguide/code-submission-process.rst [hunk 2]
```diff
@@ -1,2 +1,56 @@
...
+#. Incremental pull requests need to link to the prior iteration
+#. Incremental pull requests need to describe changes since the last PR
+#. Link to the ticket(s) that are addressed to it.
+#. When fixing an issue, update the issue status to ``In Review`` after submitting the PR.
+#. Links to the prscript builds*
+#. Pull requests are automatically tested using github actions (https://github.com/OISF/suricata/blob/master/.github/workflows/builds.yml).
+   Failing builds won't be considered and should be closed immediately.
+#. Pull requests that change, or add a feature should include a documentation update commit
+
+  (*) access to prscript is limited to trusted devs. For the rest of you, ask Victor, Eric, Andreas or Jason to run it for you.
```
Comment (norg) [inline doc/devguide/code-submission-process.rst:45]: is this list still the current one or do we have more from the team being able to run it for others?

**Comment by victorjulien** [comment]
Merged in #4777, thanks Jeff.

File: doc/devguide/code-submission-process.rst (hunk 2 above)
Comment (victorjulien) [reply doc/devguide/code-submission-process.rst:45]: This is fine for now. We're looking at replacing prscript, so we'll see what happens then.

**Pull request closed without merging**